### PR TITLE
[Metrics] Report finalized height and executed height on startup

### PR DESCRIPTION
**Pull request by zhangchiqing**
This PR fixes an issue that the finalized height and executed height metrics is not reported on startup until there is block being finalized and executed.

For EN which takes a while to load the execution state, the finalized height and executed height is not shown on metrics.

This PR fixes it by reporting finalized height and executed height on startup.

### CODE REVIEW

**Review by jordanschalm — APPROVED** [review]

**Review by lolpuddle — APPROVED** [review]

**Review by m4ksio — APPROVED** [review]

**Review by jordanschalm — COMMENTED** [review]
Suggesting some improvements to error documentation and handling.

File: storage/badger/procedure/executed.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	return func(tx *badger.Txn) error {
+		var highest flow.Header
+		err := operation.RetrieveExecutedBlock(blockID)(tx)
+		if err != nil {
+			return fmt.Errorf("could not lookup executed block %v: %w", blockID, err)
+		}
+		err = operation.RetrieveHeader(*blockID, &highest)(tx)
+		if err != nil {
+			return fmt.Errorf("could not retrieve executed header %v: %w", blockID, err)
+		}
```
[the remark this answers is not in the document]
Comment (zhangchiqing) [reply storage/badger/procedure/executed.go:48]: Thanks  for adding the comments and improving error handling 🙏

**Comment by zhangchiqing** [comment]
bors merge

**Comment by zhangchiqing** [comment]
bors merge

**Comment by zhangchiqing** [comment]
bors retry

**Comment by zhangchiqing** [comment]
bors retry

**Comment by zhangchiqing** [comment]
bors retry

**Comment by zhangchiqing** [comment]
bors retry

**Comment by zhangchiqing** [comment]
bors retry

**Comment by zhangchiqing** [comment]
bors merge

**Pull request merged**